src/lib.rs

```diff
@@ -290,17 +298,49 @@ impl Build {
             .unwrap_or_else(|| env::var("TARGET").expect("TARGET must be set"))
     }
 
-    fn find_nasm(&mut self) -> PathBuf {
-        match self.nasm.clone() {
-            Some(path) => path,
-            None => {
-                let nasm_path = PathBuf::from("nasm");
-                match is_nasm_new_enough(&nasm_path) {
-                    Ok(_) => nasm_path,
-                    Err(version) => {
-                        panic!("This version of NASM is too old: {}", version);
-                    },
+    /// Returns version string if nasm is too old,
+    /// or error message string if it's unusable.
+    fn is_nasm_new_enough(&self, nasm_path: &Path) -> Result<(), String> {
+        match get_output(Command::new(nasm_path).arg("-v")) {
+            Ok(version) => {
+                if version.contains("NASM version 0.") {
+                    Err(version)?
+                } else if let Some((major, minor, micro)) = self.min_version {
+                    let ver: Vec<u8> = version
+                        .split(" ")
+                        .skip(2)
+                        .next()
+                        .map(|ver| {
+                            ver.split(".")
+                                .map(|v| v.parse().expect("Invalid version component"))
+                                .collect()
+                        })
+                        .expect("Invalid version");
```

Comment:
There has to be a better way, I have no idea looking at this what this parsing does. My feeling is that some sort of regex may be a better solution.

Reply:
the regex is big, we can use it, but isn't exactly optimal.